src/io_object.cpp

```diff
@@ -185,10 +185,10 @@ Value IoObject::gets(Env *env) const {
     return line;
 }
 
-Value IoObject::get_path() const {
+Value IoObject::get_path() {
```

Comment:
It needs to duplicate the string, so it needs to do `m_path->as_string` (or `m_path->dup`), but then `operator->` of Value is not available for const Values, so I tried to add a const variant as well, but then hydration of the Value doesn't work anymore. At that point it felt like there were enough turtles without going all the way down.

Reply:
Thanks to the power of rubber ducky debugging, I fixed this.
The setter of path is defined like this:
```c++
void set_path(StringObject *path) { m_path = path; }
```
Since that already takes a `StringObject *` instead of a `Value`, it was pretty much trivial to change `m_path` to a `StringObject *`. This change made everything work fine.

Reply:
Oh wow, I'm sorry to send you on that chase. Good work though!